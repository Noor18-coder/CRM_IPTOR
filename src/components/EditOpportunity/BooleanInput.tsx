import React from 'react';
import * as models from '../../helpers/Api/models';

interface BooleanInputs {
  obj: models.AttributeFormField;
  onValueChange: (object: models.AttributeFormField) => void;
}

const BooleanInput: React.FC<BooleanInputs> = ({ obj, onValueChange }) => {
  const onCheckboxValueChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { checked } = e.currentTarget;
    obj.attributeValueB = checked;
    onValueChange(obj);
  };

  return (
    <>
      <div className="adding-txt-field-container">
        <div className="form-group oppty-form-elements">
          <span className="checkbox-label">
            {obj.description}
            <label className="switch" htmlFor={obj.id}>
              <input type="checkbox" tabIndex={0} checked={obj.attributeValue === 'Y'} id={obj.id} onChange={onCheckboxValueChange} />
              <span className="slider round">&nbsp;</span>
            </label>
          </span>
        </div>
      </div>
    </>
  );
};

export default BooleanInput;
