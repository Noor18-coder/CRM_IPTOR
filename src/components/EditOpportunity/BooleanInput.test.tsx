import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import BooleanInput from './BooleanInput';
import StoreMock from '../../mocks/Store.mock';
import AttributeMock from '../../mocks/Attribute.mock';

const store = StoreMock.createInitialStore();
const obj = AttributeMock.getAttributeFormFieldConstVal();
const onValueChange = jest.fn();

describe('[EditOpportunity] BooleanInput', () => {
  it('should renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BooleanInput obj={obj} onValueChange={onValueChange} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
