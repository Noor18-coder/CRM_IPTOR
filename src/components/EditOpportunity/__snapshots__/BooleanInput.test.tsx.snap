// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[EditOpportunity] BooleanInput should renders correctly 1`] = `
<div
  className="adding-txt-field-container"
>
  <div
    className="form-group oppty-form-elements"
  >
    <span
      className="checkbox-label"
    >
      Generic
      <label
        className="switch"
        htmlFor="New"
      >
        <input
          checked={false}
          id="New"
          onChange={[Function]}
          tabIndex={0}
          type="checkbox"
        />
        <span
          className="slider round"
        >
          Â 
        </span>
      </label>
    </span>
  </div>
</div>
`;
